model/jpa/src/main/java/org/keycloak/models/jpa/entities/RoleEntity.java

```diff
@@ -55,14 +55,16 @@
         @UniqueConstraint(columnNames = { "NAME", "CLIENT_REALM_CONSTRAINT" })
 })
 @NamedQueries({
-        @NamedQuery(name="getClientRoles", query="select role from RoleEntity role where role.client = :client"),
+        @NamedQuery(name="getClientRoles", query="select role from RoleEntity role where role.client.id = :client order by role.name"),
```

Comment:
I will have a look.

Reply:
Fixed, that was the conversion from stream to set which break the order. So sorting is now working in the UI and API using the new query params. But I choose to not implement it in the `/roles` without query params, because this endpoint was alreayd existing and I should not change the response because maybe user already use it with some specific script to parse it or anything.

Reply:
Ok, makes sense.